src/utils/errorSerializer.ts

```diff
@@ -34,61 +38,64 @@ import UserTimeoutError from "../errors/UserTimeoutError";
 import VersionInvalidError from "../errors/VersionInvalidError";
 import VersionMismatchError from "../errors/VersionMismatchError";
 
-const errorCodeToErrorClass: {[key: string]: any} = {
+const errorCodeToErrorClass: Record<keyof typeof ErrorCode, KnownErrorConstructors> = {
   [ErrorCode.ERR_WEBEID_ACTION_PENDING]:          ActionPendingError,
   [ErrorCode.ERR_WEBEID_ACTION_TIMEOUT]:          ActionTimeoutError,
   [ErrorCode.ERR_WEBEID_CONTEXT_INSECURE]:        ContextInsecureError,
   [ErrorCode.ERR_WEBEID_EXTENSION_UNAVAILABLE]:   ExtensionUnavailableError,
+  [ErrorCode.ERR_WEBEID_MISSING_PARAMETER]:       MissingParameterError,
   [ErrorCode.ERR_WEBEID_NATIVE_INVALID_ARGUMENT]: NativeInvalidArgumentError,
   [ErrorCode.ERR_WEBEID_NATIVE_FATAL]:            NativeFatalError,
   [ErrorCode.ERR_WEBEID_NATIVE_UNAVAILABLE]:      NativeUnavailableError,
   [ErrorCode.ERR_WEBEID_USER_CANCELLED]:          UserCancelledError,
   [ErrorCode.ERR_WEBEID_USER_TIMEOUT]:            UserTimeoutError,
   [ErrorCode.ERR_WEBEID_VERSION_INVALID]:         VersionInvalidError,
   [ErrorCode.ERR_WEBEID_VERSION_MISMATCH]:        VersionMismatchError,
+  [ErrorCode.ERR_WEBEID_UNKNOWN_ERROR]:           UnknownError,
 };
 
-export function serializeError(error: any): any {
-  const {
-    message,
-    name,
-    fileName,
-    lineNumber,
-    columnNumber,
-    stack,
-  } = error;
+export function serializeError(error: Error): SerializedError {
+  if (typeof error != "object" || error == null || !(error instanceof Error)) {
+    throw new UnknownError();
+  }
+
+  const result: Record<string, unknown> = {};
+
+  for (const prop of Object.getOwnPropertyNames(error)) {
+    Reflect.set(result, prop, Reflect.get(error, prop));
+  }
+
+  return result as SerializedError;
+}
 
-  return {
-    ...(
-      Object.fromEntries(
-        Object.getOwnPropertyNames(error)
-          .map((prop) => [prop, error[prop]])
-      )
-    ),
+function isSerializedError(obj: unknown): obj is SerializedError {
+  if (typeof obj !== "object" || obj === null) {
+    return false;
+  }
 
-    message,
-    name,
-    fileName,
-    lineNumber,
-    columnNumber,
-    stack,
-  };
+  const errorObject = obj as SerializedError;
+
+  return (
+    typeof errorObject.name    === "string" &&
+    typeof errorObject.message === "string" &&
+    typeof errorObject.stack   === "string"
```

Comment:
Do you want to check `errorObject.code` here as well? You are using it in `deserializeError()` line 89 after `isSerializedError()`.

Reply:
Good catch. I was thinking the usual Error object doesn't have the `code` property, but all serialized errors will be custom errors and will have the `code` property.